WordPress/gutenberg
# Compat: Upgrade admin notices to use Notices module at runtime

Related: #5975, #6388
Supersedes #5590 
Closes #10448 

This pull request seeks to try to seamlessly upgrade notices output via an `admin_notices` or `all_admin_notices` action server-side.

**Implementation notes:**

The flow here is:

- Discover admin notices by `.notice` class
- Convert to `@wordpress/notices` notice object
   - This occurs in `edit-post`, though in the future I think either this compatibility should (a) become unnecessary or (b) be moved somewhere more generic
- Dispatch to create notice through `@wordpress/notices`
- Removes the admin notice from the DOM

**Testing instructions:**

Here's a plugin if you need one:

[demo-notice.zip](https://github.com/WordPress/gutenberg/files/2559361/demo-notice.zip)

This will add a dismissible success notice to any admin screen.

Verify that upon visiting Gutenberg, a notice echo'd during `admin_notices` or `all_admin_notices` is displayed.

**Pending tasks:**

- [x] Notice content is supported only as a plaintext string. Advanced markup is not supported. In not supporting non-plaintext, we can't truly close #10448, which includes an inline link. The notices module supports linkable "actions", but not interspersed with the plaintext. This ties in a bit with #9846, with the most direct solutions being one of either (a) allow elements as notice content or (b) pass notice content to a [`RawHTML` component](https://github.com/WordPress/gutenberg/blob/master/packages/element/src/raw-html.js), neither of which is ideal in my mind.
